packages/test/test-end-to-end-tests/src/test/gc/gcContainerRuntimeCompat.spec.ts

```diff
@@ -106,35 +91,82 @@ describeFullCompat("GC summary compatibility tests", (getTestObjectProvider) =>
+        assert(
+            latestSummaryContext
+                && latestSummaryContext.referenceSequenceNumber >= summary.summarySequenceNumber,
+            `Did not get expected summary. Expected: ${summary.summarySequenceNumber}. ` +
+            `Actual: ${latestSummaryContext?.referenceSequenceNumber}.`,
+        );
+        assert(latestUploadedSummary !== undefined, "Did not get a summary");
+
+        const gcState = getGCStateFromSummary(latestUploadedSummary);
+        assert(gcState !== undefined, "GC tree is not available in the summary");
+        const nodeTimestamps: Map<string, number | undefined> = new Map();
+        for (const [nodeId, nodeData] of Object.entries(gcState.gcNodes)) {
+            nodeTimestamps.set(nodeId.slice(1), nodeData.unreferencedTimestampMs);
+        }
+        return nodeTimestamps;
+    }
+
+    /**
+     * Creates and returns the container runtime factory and data store factory with the given version.
      * @param version - Can be 0, -1 or -2 where 0 represents current version and -1 and -2 represent older versions.
      */
-    function getContainerRuntimeFactory(version: number) {
+    function createRuntimeFactories(version: number) {
         const containerRuntimeApi = getContainerRuntimeApi(pkgVersion, version);
-        return new containerRuntimeApi.ContainerRuntimeFactoryWithDefaultDataStore(
-            dataObjectFactory,
+        const dataRuntimeApi = getDataRuntimeApi(pkgVersion, version);
+        const dataStoreFactory = new dataRuntimeApi.DataObjectFactory(
+            "TestDataObject",
+            getTestFluidObjectType(dataRuntimeApi.DataObject),
+            [],
+            [],
+        );
+        const innerRequestHandler = async (request: IRequest, runtime: IContainerRuntimeBase) =>
+            runtime.IFluidHandleContext.resolveHandle(request);
+        const containerRuntimeFactory = new containerRuntimeApi.ContainerRuntimeFactoryWithDefaultDataStore(
+            dataStoreFactory,
             [
-                [dataObjectFactory.type, Promise.resolve(dataObjectFactory)],
+                [dataStoreFactory.type, Promise.resolve(dataStoreFactory)],
             ],
             undefined,
             [innerRequestHandler],
-            runtimeOptions,
+            {
+                summaryOptions: { disableSummaries: true },
+                gcOptions: { gcAllowed: true, writeDataAtRoot: true },
+            },
         );
+        return { containerRuntimeFactory, dataStoreFactory };
     }
 
-    beforeEach(async () => {
+    before(async function() {
+        // The default timeout for before block completion is 5 seconds. However, it usually takes more than that to
+        // install packages. Set the timeout to 20 seconds which should be enough to install the required packages.
+        this.timeout(20000);
+
+        // Ensure that the packages used by this test are installed.
+        await ensurePackageInstalled(pkgVersion, 0, false /* boolean */);
```

Comment:
@curtisman Can you please confirm that this is the correct usage :-). The test installs the versions it uses in the before block.

Reply:
Currently all e2e test are written for a specific combination of versions.   Can your test be written a way that will honor those combinations so that flags like --compatVersion and --compatKind would work?

Otherwise, your test will be run multiple times when the pipeline split the compat combinations into parallel runs.

Reply:
I see. Let me take a look. The tests will need to know about what the versions are. Maybe these have to be exposed via the test object provider.

Reply:
Is this conversation resolved?

Reply:
Nope, I haven't worked on it. I will update here once I do that.